src/BTVNanoCommissioning/workflows/QCD_soft_mu_validation.py

```diff
@@ -0,0 +1,524 @@
+            # same_length_events = filtered_events[equal_length_events]
+            # print("Number of events with equal number of Jet and JetSVs:", len(same_length_events))
+            # print("Same length selev.Jet", same_length_events.Jet)
+            # print("Same length selev.JetSVs.jetIdx", same_length_events.JetSVs.jetIdx)
+
+            # Count and print the number of events where JetSVs is longer than Jet
+            # longer_jetSVs_events = ak.count(events.JetSVs.pt, axis=1) > ak.count(events.Jet.pt, axis=1)
+            # longer_jetSVs_events = longer_jetSVs_events & (ak.count(events.Jet.pt, axis=1) == 0)
+            # num_longer_jetSVs_events = ak.sum(longer_jetSVs_events)
+            # print("Number of events where JetSVs is longer than Jet:", num_longer_jetSVs_events)
+            # print("Longer SV selev.Jet", events.Jet[longer_jetSVs_events])
+            # print("Longer SV selev.JetSVs.jetIdx", events.JetSVs.jetIdx[longer_jetSVs_events])
+
+            # Ensure that all indices are within the valid range for selev.Jet
+            valid_indices = (filtered_events.JetSVs.jetIdx >= 0) & (
+                filtered_events.JetSVs.jetIdx < ak.num(filtered_events.Jet)
+            )
+            if not np.all(valid_indices):
+                print("Warning: Some indices in selev.JetSVs.jetIdx are out of range.")
+                # Filter out invalid indices
+                filtered_events.JetSVs = filtered_events.JetSVs[valid_indices]
+
+            # Check if selev.JetSVs.jetIdx is empty after filtering
+            if len(filtered_events.JetSVs.jetIdx) == 0:
+                print("Warning: selev.JetSVs.jetIdx is empty after filtering.")
+                matched_JetSVs = ak.Array([])
+                lj_matched_JetSVs = ak.Array([])
+                lj_SVs = ak.Array([])
+                nJetSVs = ak.Array([])
+            else:
+                # Proceed with the assignment
+                matched_JetSVs = filtered_events.Jet[filtered_events.JetSVs.jetIdx]
+                lj_matched_JetSVs = matched_JetSVs[filtered_events.JetSVs.jetIdx == 0]
+                lj_SVs = filtered_events.JetSVs[filtered_events.JetSVs.jetIdx == 0]
+                nJetSVs = ak.count(lj_SVs.pt, axis=1)
+
+            # Print the final state of the variables
+            print("matched_JetSVs:", matched_JetSVs)
+            print("lj_matched_JetSVs:", lj_matched_JetSVs)
+            print("lj_SVs:", lj_SVs)
+            print("nJetSVs:", nJetSVs)
+
+        ####################
+        # Weight & Geninfo #
+        ####################
+        weights = Weights(len(selev), storeIndividual=True)
+        if not isRealData and len(selev) > 0:
+            if len(selev.genWeight) > 0:
+                weights.add("genweight", selev.genWeight)
+            else:
+                print("Warning: selev.genWeight is empty, skipping weight addition.")

```

Comment:
after L193, the `len(selev)==0` should thrown away, why you still need to check here?

Reply:
I guess it was a security measure that I included after several MC files failed.

Reply:
I think would be good to find out why failures happens :p